lib/pool.js

```diff
@@ -0,0 +1,93 @@
+
+const debug                = require('debug')('kuss:pool')
+const R                    = require('ramda')
+const Bluebird             = require('bluebird')
+const GenericPool          = require('generic-pool')
+const { FibonacciBackoff } = require('simple-backoff')
+
+const CONFIG_PROP_SERVER    = 'server'
+const CONFIG_PROP_VALIDATOR = 'validateAsync'
+const CONFIG_PROP_TEST      = 'testOnBorrow'
+
+const DEFAULT_CONFIG = {
+  max               : 10
+, min               : 2
+, idleTimeoutMillis : 30000
+, log               : false
+, testOnBorrow      : false
+, logger            : console
+}
+
+
+const shutdown = (pool) => () => pool.drain().then(() => pool.clear())
+
+
+const getServerConfig = R.propOr({}, CONFIG_PROP_SERVER)
+
+
+const addBackoff = (factory) => {
+  const backoff   = new FibonacciBackoff()
+  const orgCreate = factory.create
```

Comment:
The original create method.  I'm following a pattern found in the express.js code